pgmpy/tests/test_metrics/test_metrics.py

```diff
@@ -224,3 +226,123 @@ def test_shd_unequal_graphs(self):
 
         with self.assertRaises(ValueError, msg="The graphs must have the same nodes."):
             SHD(dag1, dag2)
+
+
+class TestGraphicalSelfCompatibility(unittest.TestCase):
```

Comment:
For the tests, I would suggest the following:
1. Write tests for the _latent_admg method. Give it some known DAG and check if the output is correct.
2. For the compatibility score: a) use a fake estimator to give the correct sub-DAG for each estimator, and check if the incompatibility score is 0. b) Use a known model from example_models, simulate some data from it, and check the incompatibility score; it should be low as the data has been simulated from the model.